src/subscription/subscribe.js

```diff
@@ -220,16 +249,23 @@ export function createSourceEventStream(
     info
   );
 
-  if (subscription instanceof Error) {
-    throw subscription;
-  }
+  return Promise.resolve(subscription).then(resolvedSubscription => {
```

Comment:
Here's what the code looks like using a wrapped promise: https://gist.github.com/robzhu/d2d9e3238b586e67569646a5944154b8